src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/entity/TimelineEventEntity.kt

```diff
@@ -0,0 +1,101 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.EnumType
+import jakarta.persistence.Enumerated
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.Id
+import jakarta.persistence.Table
+import jakarta.validation.constraints.NotNull
+import org.hibernate.Hibernate
+import org.hibernate.annotations.CreationTimestamp
+import org.hibernate.annotations.UuidGenerator
+import java.time.Instant
+import java.util.UUID
+
+@Table(name = "timeline_event")
+@Entity
+class TimelineEventEntity(
+  @Id
+  @GeneratedValue
+  @UuidGenerator
+  var id: UUID? = null,
+
+  @Column(updatable = false)
+  @field:NotNull
+  var reference: UUID? = null,
+
+  @Column(updatable = false)
+  @field:NotNull
+  var sourceReference: String? = null,
+
+  @Column(updatable = false)
+  @Enumerated(value = EnumType.STRING)
+  @field:NotNull
+  var eventType: TimelineEventType? = null,
+
+  @Column(updatable = false)
+  var contextualInfo: String? = null,
+
+  /**
+   * The ID of the prison that the event relates to.
+   */
+  @Column(updatable = false)
+  @field:NotNull
+  var createdAtPrison: String? = null,
+
+  /**
+   * Username of the person who created the original event (not to be confused with who/what saved this entity)
```

Comment:
I've added comments to some of these fields, as they're easily confused with fields of the same name in other entities. They're effectively the same, but the intention is different and they're populated manually from the original action/entity, rather than via JPA annotations, such as `@CreatedBy`. I could consider renaming them.